### ISSUE TITLE
[stable/external-dns] Azure private dns failed to refresh the token

### ISSUE THREAD

**Issue by IbraheemAlSaady** [helm/charts#20212]
I'm using azure private DNS and I'm trying to make external DNS work with it. But it's failing to get the token

```yml
provider: azure-private-dns

azure:
  resourceGroup:
  tenantId:
  subscriptionId:
  aadClientId:
  aadClientSecret:
```

Of course its filled with values. But I'm getting this error:

`Failed to refresh the Token for request to https://management.azure.com/subscriptions//resourceGroups//providers/Microsoft.Network/privateDnsZones?api-version=2018-09-01`

The subscription id and the resource group are not being picked up.

These are my chart info: 

```terraform
image = "bitnami/external-dns"
tag = "0.5.18"
chart_version = "2.14.3"
```

Am I missing something?

Cheers!

**Comment by juan131** [comment]
Hi @IbraheemAlSaady 

Could you try setting the parameter `azure.secretName` including your `azure.json` file?

```console
$ kubectl create secret generic azure-config-file --from-file=/local/path/to/azure.json
```

You can find more information in the links below:

- https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md#creating-configuration-file
- https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/azure.md#creating-service-principal

**Comment by IbraheemAlSaady** [comment]
@juan131, I already did, but it says it'll be deprecated on the values.yml. 
I created anyway, but it still gives me that error.

**Comment by juan131** [comment]
Hi @IbraheemAlSaady 

You're totally right, that if statement is incomplete. Thanks for reporting it!

Please feel free to create a PR to address it and I'll be glad to review it. Otherwise, I can create the PR too (but I'd rather you to contribute with the project)

**Comment by IbraheemAlSaady** [comment]
@juan131

Great, I'm on it

**Comment by saidst** [comment]
You are right, as part of developing the private dns provider, we have _**not**_ updated the helm-charts so far.
The Azure Private DNS Provider only takes environment-args.
Thanks for working on that.

**Comment by IbraheemAlSaady** [comment]
@juan131 Please check for review [PR-20276](https://github.com/helm/charts/pull/20276)

**Comment by juan131** [comment]
Thanks so much for the PR @IbraheemAlSaady !!
My partner @sameersbn is already reviewing it

**Comment by IbraheemAlSaady** [comment]
I will close this issue as my changes are merged now

It's available at chart version `2.15.1`
